blog/content/post/2020-04-03-trenchboot-nlnet-lz-validation.md

```diff
@@ -0,0 +1,796 @@
+want to perform any operations step-by-step, I strongly recommend to use exactly
+the same setup.
+
+At this stage, I assume you have already installed NixOS according to our
+[instructions](https://blog.3mdeb.com/2020/2020-03-31-trenchboot-nlnet-lz/#nixos-installation).
+If yes, then we can start process of **enabling DRTM**.
+
+### Qucik reminder - boot flow
+
+Before we execute essential part, I would like to pass through quick reminder.
+Don't worry, I won't give a lecture or bring unfamiliar concepts. I want to show
+you differences between platform without DRTM and with DRTM. It is presented in
+graphical form, so it will be easier to visualize it.
+
+A normal boot process without DRTM enabled, typically looks like this:
+
+![boot without drtm](../../static/img/boot-non-drtm.png)
+
+GRUB bootloader **directly** loads Linux kernel and whereby NixOS boots. As you
+know, by modifying GRUB, Linux kernel and adding special boot modules, we
+enable DRTM. Then, the boot process looks like this:
+
+![boot drtm](../../static/img/boot-drtm.png)
+
+You can clearly see 3 additional elements, which are **slaunch module**,
+**SKINIT** and **Landing Zone**. Their functionality should be already known by
+you. Briefly, it is DRTM stage, when platform boots.
+
+Now, When you can visualize differences, you will learn how to switch your
+platform from first case to second one. Let's do it!
+
+## System customization - enabling DRTM
+
+It is the first thing we need to do. Clean NixOS doesn't meet all requirements.
+First of all, we want to replace default `nixpkgs` with our custom one. Second
+them.  of all, as it doesn't have all necessary packages by default, we want to
+install them.
+
+Fortunately, customization of NixOS will demand its **configuration update, but
+not entire system re-installation**. Moreover, there is no need to install every
+single package manually. Of course, it still can be done if you wish. However,
+entire process of enabling DRTM is automated by us to minimize user's effort. So
+much for the introduction - now we can finally run the procedure. Boot to NixOS
+and follow these steps:
+
+1. Install `cachix`
+
+`Cachix` is binary cache hosting. It allows to store binary files, so there is
+no need to build them on your own. If it is not very useful for small builds, it
+is very handy for large ones e.g. Linux kernel binary.
+
+    ```bash
+    $ nix-env -iA cachix -f https://cachix.org/api/v1/install
+    ```
+
+2. Add 3mdeb cachix hosting as default.
+
+    ```bash
+    $ cachix use 3mdeb
+    Cachix configuration written to /etc/nixos/cachix.nix.
+    Binary cache 3mdeb configuration written to /etc/nixos/cachix/3mdeb.nix.
+
+    To start using cachix add the following to your /etc/nixos/configuration.nix:
+
+        imports = [ ./cachix.nix ];
+
+    Then run:
+
+        $ sudo nixos-rebuild switch
+    ```
+
+> There is no need now to edit /etc/nixos/configuration.nix as above log
```

Comment:
@piotr-kleins That's not true. It's not included in the basic configuration and user needs to add it on his own.

Reply:
Removed this and added information to edit `configuration.nix`